src/main/java/jenkins/plugins/gerrit/workflow/GerritReviewStep.java

```diff
@@ -64,10 +68,27 @@ protected Void run() throws Exception {
           int changeId = Integer.parseInt(matcher.group("changeId"));
           int revision = Integer.parseInt(matcher.group("revision"));
           listener.getLogger().format("Gerrit review change %d/%d label %s=%d (%s)%n", changeId, revision, label, score, message);
-          ReviewInput reviewInput = new ReviewInput().label(label, score).message(message);
+          ReviewInput reviewInput = new ReviewInput().message(message);
+          boolean notifyOwner = false;
+          if (labels == null) {
+            if (label != null) {
+                reviewInput.label(label, score);
+                if (score < 0) {
+                  notifyOwner = true;
+                }
+            }
+          }
+          else {
+            for (Map.Entry<String, Integer> l : labels.entrySet()) {
+              reviewInput.label(l.getKey(), l.getValue());
+              if (l.getValue() < 0) {
+                notifyOwner = true;
+              }
+            }
+          }
           reviewInput.drafts = ReviewInput.DraftHandling.PUBLISH;
-          reviewInput.tag = String.format("autogenerated:jenkins:%s", label);
-          if (score < 0) {
+          reviewInput.tag = "autogenerated:jenkins";
```

Comment:
You sure we don't need the label in the tag? I literally copied on how it is in the Gerrit-CI at the moment and I had to reverse engineer it from what Gerrit does.

Reply:
it looks like the tag is only information, not doing anything... notice we do not have tag for the comment. I thought of removing both or adding both. even in the gerrit restapi the tag has no method, so I am unsure it is relevant. in any case it works for me...